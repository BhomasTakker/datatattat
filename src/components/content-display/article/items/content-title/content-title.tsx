import { TitleVariant } from "@/src/components/types/ui";
import { Title } from "@/src/components/ui/title";
import { CSS } from "@/src/css/text";
import { log } from "@/src/lib/logger";
import { Typography } from "@mui/material";
import { MARGINS } from "config/styles/styles.config";

export type ContentTitleVariants = "Primary" | "Secondary";

// And element props?
export type ContentTitleProps = {
	title: string;
	variant?: ContentTitleVariants;
	maxLines?: number;
};

export const ContentTitle = ({
	title,
	variant = "Primary",
	maxLines = 2,
}: ContentTitleProps) => {
	log({ code: "FEATURE:0010", context: "CONTENT:TITLE" }, { variant });
	return (
		<Typography
			// variant compact
			variant="h6"
			component="h6"
			marginTop={0}
			marginBottom={0}
			///////////////////
			sx={{
				...CSS.maxLines({ maxLines }),
			}}
			// {...CSS.maxLines({ maxLines })}
		>
			{title}
		</Typography>
	);
};
