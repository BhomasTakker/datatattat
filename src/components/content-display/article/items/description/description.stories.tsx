import type { Meta, StoryObj } from "@storybook/react";
import { Description } from "./description";
import { LOREM_5_S } from "mockData/text/lorem";

const meta: Meta<typeof Description> = {
	component: Description,
};
export default meta;
type Story = StoryObj<typeof Description>;

export const Standard: Story = {
	args: {
		description: LOREM_5_S,
		maxLines: 3,
		maxWidth: "100%",
		minWidth: "400px",
	},
};
