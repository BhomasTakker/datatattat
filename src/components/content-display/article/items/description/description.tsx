import { Typography } from "@mui/material";
import { CSS } from "@/src/css/text";
import { stripHTML } from "@/src/utils/html";
import { log } from "@/src/lib/logger";
import { TypographyProps } from "@/src/types/mui";

export type DescriptionVariants = "Primary" | "Secondary";

export type DescriptionProps = {
	description: string;
	maxLines?: number;
	// maxWidth?: string; // how to do number px | % etc
	// minWidth?: string;
	// variant: DescriptionVariants;
} & TypographyProps;

export const Description = ({
	description,
	maxLines = 3,
	// variant = "Primary",
	// look at this - we probably don't want defaults?
	// maxWidth = "100%",
	// 400px AND 100% to be taken from a widths
	// i.e. SIZES.FULL, SIZES.PRESET_400
	// minWidth = "400px",
	...rest
}: DescriptionProps) => {
	log({ code: "FEATURE:0010", context: "DESCRIPTION" });
	return (
		<Typography
			data-testid="description"
			sx={{
				// width object / max, min, breakpoint sizes?
				// ...minMaxWidth({ maxWidth, minWidth }), // do a width object
				...CSS.maxLines({ maxLines }),
			}}
			{...rest}
			// This would be a no? you could go with text variant
			// font family, size, color, etc
			{...CSS.textStyle()}
		>
			{stripHTML(description)}
		</Typography>
	);
};
