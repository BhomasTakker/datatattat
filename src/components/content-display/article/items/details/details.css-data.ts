import { StackProps, StackTypeMap } from "@mui/material";
import { MARGINS } from "config/styles/styles.config";

export const spaceBetween: StackProps = {
	direction: "row",
	justifyContent: "space-between",
	marginTop: MARGINS.SMALL,
};

export const stack: StackProps = {
	direction: "column",
	marginTop: MARGINS.SMALL,
};

export const DetailsVariant = new Map([
	["space-between", spaceBetween],
	["stack", stack],
]);
