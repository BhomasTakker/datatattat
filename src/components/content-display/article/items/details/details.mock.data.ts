export const details1 = {
	docs: [],
	categories: ["category1", "category2"],
	authors: ["Author McAuther", "Authorson, Author"],
	publishers: ["Publisher.com"],
	published: "Tue, 24 Oct 2023 13:30:00 +0100",
	modified: "Tue, 25 Oct 2023 13:30:00 +0100",
};
