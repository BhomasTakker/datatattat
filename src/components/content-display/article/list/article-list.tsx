import { Collection } from "@/src/types/data-structures/collection/collection";
import { List, ListProps } from "@mui/material";
import { ArticleListItemController } from "../items/list-item/article-list-item.controller";
import { ArticleListItemVariant } from "../items/list-item/article-list-item.config";

export type ArticleListProps = {
	data: Collection;
	variant?: ArticleListItemVariant;
	limit?: number;
} & ListProps;

export const ArticleList = ({
	data,
	children,
	variant = "expanded",
	limit,
	// we want to have overflow
	...rest
}: ArticleListProps) => {
	const { items = [] } = data;

	const renderItems = items.slice(0, limit || items.length);

	// Should we be showing title etc?
	// I expect that's better as a wrapper but maybe?
	// Probably is better here - greater control
	// just create a component

	// Use default or provided list
	const renderChildren = children
		? children
		: renderItems.map((item, i) => (
				// argument to clone the data?
				// Should we be loading a wrapper?
				<ArticleListItemController
					key={item.title}
					item={item}
					variant={variant}
				/>
		  ));

	return (
		<List {...rest} data-testid="article-list">
			{renderChildren}
		</List>
	);
};
