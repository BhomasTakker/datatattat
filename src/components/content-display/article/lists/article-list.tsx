interface ArticleListProps {
	queryData: any; // should be data
	component: string;
	variant: string;
}

// 100% change queryData to data
export const ArticleList = ({
	queryData,
	component,
	variant,
}: ArticleListProps) => {
	// Now need look at Query better
	console.log("FEATURE:001", "ARTICLE:LIST", { queryData });
	console.log("FEATURE:001", "ARTICLE:LIST", { component });
	console.log("FEATURE:001", "ARTICLE:LIST", { variant });

	return <>ArticleList</>;
};
