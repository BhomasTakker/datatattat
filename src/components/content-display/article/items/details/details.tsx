import { Stack, Typography } from "@mui/material";
import { MARGINS } from "config/styles/styles.config";
import { Time } from "../time/time";
import { Details } from "@/src/types/data-structures/base";
import { DetailsVariant } from "./details.css-data";

interface DetailsTextProps {
	children: React.ReactNode;
}

const DetailsText = ({ children }: DetailsTextProps) => {
	return <Typography>{children}</Typography>;
};

type DetailsVariantType = "space-between" | "stack";

interface DetailsComponentProps {
	details?: Details;
	showPublished?: boolean;
	showCategories?: boolean;
	showpublishers?: boolean;
	showAuthors?: boolean;
	variant?: DetailsVariantType;
	// divider?: boolean;
}

export const DetailsComponent = ({
	details,
	showPublished = false,
	showCategories = false,
	showpublishers = false,
	showAuthors = false,
	// divider = true,
	variant = "space-between",
}: DetailsComponentProps) => {
	const {
		published = null,
		categories = null,
		publishers = null,
		authors = null,
	} = details || {};
	console.log("FETURE:0007", "Details", {
		variant,
		DetailsVariant,
	});
	const variantProps = DetailsVariant.get(variant);
	console.log("FETURE:0007", "Details", {
		variantProps,
	});
	console.log("FETURE:0007", "DetailsVariant", {
		DetailsVariant,
	});
	return (
		<Stack
			data-testid="details"
			{...variantProps}
			// divider={React Element}
		>
			{/* do something proper with the time!! */}
			{/* Like we want an image AND avatar component - we want Time - variant age/publish date */}
			{showPublished && <Time time={published} variant="age" />}
			{showCategories && <DetailsText>{categories?.join(", ")}</DetailsText>}
			{showpublishers && <DetailsText>{publishers?.join(", ")}</DetailsText>}
			{showAuthors && <DetailsText>{authors?.join(", ")}</DetailsText>}
		</Stack>
	);
};
