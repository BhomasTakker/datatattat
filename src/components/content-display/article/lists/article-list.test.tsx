import { render, screen } from "@testing-library/react";
import { ArticleList, ArticleListProps } from "./article-list";
import { LOREM_1_S } from "mockData/text/lorem";
import { ContentTitleVariants } from "../items/content-title/content-title";
import { DescriptionVariants } from "../items/description/description";
import { DetailsVariantType } from "../items/details/details";
import { Articles } from "../mock/Articles.mock";

const props: ArticleListProps = {
	data: Articles,
	// title: "Test Title",
	// description: LOREM_1_S,
	// component,
	useAvatar: true,
	showDescription: true,
	showPublished: true,
	showAuthor: true,
	showPublisher: true,
	///////////
	componentTitle: "Test Title",
	componentTitleVariant: "Primary" as ContentTitleVariants,
	componentDescription: LOREM_1_S,
	componentDescriptionVariant: "Primary" as DescriptionVariants,
	itemTitleVariant: "Primary" as ContentTitleVariants,
	itemTitleMaxLines: 0,
	itemDescriptionVariant: "Primary" as DescriptionVariants,
	itemDescriptionMaxLines: 0,
	itemDetailsVariant: "space-between" as DetailsVariantType,
};

const { componentTitle } = props;

describe("ArticleList", () => {
	it("should render article list item", () => {
		render(<ArticleList {...props} />);
		const articleListItem = screen.getByTestId("article-list");
		expect(articleListItem).toBeInTheDocument();
	});

	it("should render expected title", () => {
		render(<ArticleList {...props} />);
		const articleListItem = screen.getByText(componentTitle);
		expect(articleListItem).toBeInTheDocument();
	});

	it("should render expected number of items", () => {
		const expectedLength = Articles.items.length;
		render(<ArticleList {...props} />);
		const articleListItems = screen.getAllByTestId("article-list-item");
		expect(articleListItems.length).toBe(expectedLength);
	});

	describe("snaps", () => {
		it("should match snapshot for article list item", () => {
			render(<ArticleList {...props} />);
			const articleListItem = screen.getByTestId("article-list");
			expect(articleListItem).toMatchSnapshot();
		});
	});
});
