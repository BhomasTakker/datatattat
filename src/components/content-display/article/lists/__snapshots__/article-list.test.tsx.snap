// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList snaps should match snapshot for article list item 1`] = `
<div
  class="MuiBox-root css-0"
  data-testid="article-list"
>
  <h6
    class="MuiTypography-root MuiTypography-h6 css-1ecob9d-MuiTypography-root"
  >
    Test Title
  </h6>
  <ul
    class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      data-testid="article-list-item"
    >
      <div
        class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
          data-testid="avatar"
        >
          <img
            alt="Dwayne Johnson 'belly laughs' after controversial waxwork unveiled"
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="https://e3.365dm.com/23/10/70x70/skynews-dwayne-johnson-waxwork_6333809.jpg?20231024132712"
          />
        </div>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-asrneu-MuiTypography-root"
        >
          Dwayne Johnson 'belly laughs' after controversial waxwork unveiled
        </h6>
        <div
          class="css-urd5lq-MuiStack-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1etgu4m-MuiTypography-root"
            data-testid="description"
          >
            A museum has changed the skin tone of its Dwayne Johnson wax statue after it was criticised for being too white.
          </span>
          <div
            class="css-1jligzp-MuiStack-root"
            data-testid="details"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
              data-testid="time"
            >
              Published 5 days ago
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Publisher1, Publisher1
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Author1, Author2
            </p>
          </div>
        </div>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      data-testid="article-list-item"
    >
      <div
        class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
          data-testid="avatar"
        >
          <img
            alt="Not Dwayne Johnson 'belly laughs' after controversial waxwork unveiled"
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="https://e3.365dm.com/23/10/70x70/skynews-dwayne-johnson-waxwork_6333809.jpg?20231024132712"
          />
        </div>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-asrneu-MuiTypography-root"
        >
          Not Dwayne Johnson 'belly laughs' after controversial waxwork unveiled
        </h6>
        <div
          class="css-urd5lq-MuiStack-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1etgu4m-MuiTypography-root"
            data-testid="description"
          >
            A museum has changed the skin tone of its Dwayne Johnson wax statue after it was criticised for being too white.
          </span>
          <div
            class="css-1jligzp-MuiStack-root"
            data-testid="details"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
              data-testid="time"
            >
              Published 1 hour ago
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Publisher1, Publisher1
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Author1, Author2
            </p>
          </div>
        </div>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      data-testid="article-list-item"
    >
      <div
        class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
          data-testid="avatar"
        >
          <img
            alt="Also Not Dwayne Johnson 'belly laughs' after controversial waxwork unveiled"
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="https://e3.365dm.com/23/10/70x70/skynews-dwayne-johnson-waxwork_6333809.jpg?20231024132712"
          />
        </div>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-asrneu-MuiTypography-root"
        >
          Also Not Dwayne Johnson 'belly laughs' after controversial waxwork unveiled
        </h6>
        <div
          class="css-urd5lq-MuiStack-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1etgu4m-MuiTypography-root"
            data-testid="description"
          >
            A museum has changed the skin tone of its Dwayne Johnson wax statue after it was criticised for being too white.
          </span>
          <div
            class="css-1jligzp-MuiStack-root"
            data-testid="details"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
              data-testid="time"
            >
              Published 1 week ago
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Publisher1, Publisher1
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Author1, Author2
            </p>
          </div>
        </div>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      data-testid="article-list-item"
    >
      <div
        class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
          data-testid="avatar"
        >
          <img
            alt="Dwayne Johnson 'belly laughs' after controversial waxwork unveiled"
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="https://e3.365dm.com/23/10/70x70/skynews-dwayne-johnson-waxwork_6333809.jpg?20231024132712"
          />
        </div>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-asrneu-MuiTypography-root"
        >
          Dwayne Johnson 'belly laughs' after controversial waxwork unveiled
        </h6>
        <div
          class="css-urd5lq-MuiStack-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1etgu4m-MuiTypography-root"
            data-testid="description"
          >
            A museum has changed the skin tone of its Dwayne Johnson wax statue after it was criticised for being too white.
          </span>
          <div
            class="css-1jligzp-MuiStack-root"
            data-testid="details"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
              data-testid="time"
            >
              Published 5 days ago
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Publisher1, Publisher1
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Author1, Author2
            </p>
          </div>
        </div>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      data-testid="article-list-item"
    >
      <div
        class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
          data-testid="avatar"
        >
          <img
            alt="Not Dwayne Johnson 'belly laughs' after controversial waxwork unveiled"
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="https://e3.365dm.com/23/10/70x70/skynews-dwayne-johnson-waxwork_6333809.jpg?20231024132712"
          />
        </div>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-asrneu-MuiTypography-root"
        >
          Not Dwayne Johnson 'belly laughs' after controversial waxwork unveiled
        </h6>
        <div
          class="css-urd5lq-MuiStack-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1etgu4m-MuiTypography-root"
            data-testid="description"
          >
            A museum has changed the skin tone of its Dwayne Johnson wax statue after it was criticised for being too white.
          </span>
          <div
            class="css-1jligzp-MuiStack-root"
            data-testid="details"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
              data-testid="time"
            >
              Published 1 hour ago
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Publisher1, Publisher1
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Author1, Author2
            </p>
          </div>
        </div>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      data-testid="article-list-item"
    >
      <div
        class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
          data-testid="avatar"
        >
          <img
            alt="Also Not Dwayne Johnson 'belly laughs' after controversial waxwork unveiled"
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="https://e3.365dm.com/23/10/70x70/skynews-dwayne-johnson-waxwork_6333809.jpg?20231024132712"
          />
        </div>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-asrneu-MuiTypography-root"
        >
          Also Not Dwayne Johnson 'belly laughs' after controversial waxwork unveiled
        </h6>
        <div
          class="css-urd5lq-MuiStack-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1etgu4m-MuiTypography-root"
            data-testid="description"
          >
            A museum has changed the skin tone of its Dwayne Johnson wax statue after it was criticised for being too white.
          </span>
          <div
            class="css-1jligzp-MuiStack-root"
            data-testid="details"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
              data-testid="time"
            >
              Published 1 week ago
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Publisher1, Publisher1
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Author1, Author2
            </p>
          </div>
        </div>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      data-testid="article-list-item"
    >
      <div
        class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
          data-testid="avatar"
        >
          <img
            alt="Dwayne Johnson 'belly laughs' after controversial waxwork unveiled"
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="https://e3.365dm.com/23/10/70x70/skynews-dwayne-johnson-waxwork_6333809.jpg?20231024132712"
          />
        </div>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-asrneu-MuiTypography-root"
        >
          Dwayne Johnson 'belly laughs' after controversial waxwork unveiled
        </h6>
        <div
          class="css-urd5lq-MuiStack-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1etgu4m-MuiTypography-root"
            data-testid="description"
          >
            A museum has changed the skin tone of its Dwayne Johnson wax statue after it was criticised for being too white.
          </span>
          <div
            class="css-1jligzp-MuiStack-root"
            data-testid="details"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
              data-testid="time"
            >
              Published 5 days ago
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Publisher1, Publisher1
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Author1, Author2
            </p>
          </div>
        </div>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      data-testid="article-list-item"
    >
      <div
        class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
          data-testid="avatar"
        >
          <img
            alt="Not Dwayne Johnson 'belly laughs' after controversial waxwork unveiled"
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="https://e3.365dm.com/23/10/70x70/skynews-dwayne-johnson-waxwork_6333809.jpg?20231024132712"
          />
        </div>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-asrneu-MuiTypography-root"
        >
          Not Dwayne Johnson 'belly laughs' after controversial waxwork unveiled
        </h6>
        <div
          class="css-urd5lq-MuiStack-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1etgu4m-MuiTypography-root"
            data-testid="description"
          >
            A museum has changed the skin tone of its Dwayne Johnson wax statue after it was criticised for being too white.
          </span>
          <div
            class="css-1jligzp-MuiStack-root"
            data-testid="details"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
              data-testid="time"
            >
              Published 1 hour ago
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Publisher1, Publisher1
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Author1, Author2
            </p>
          </div>
        </div>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      data-testid="article-list-item"
    >
      <div
        class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
          data-testid="avatar"
        >
          <img
            alt="Also Not Dwayne Johnson 'belly laughs' after controversial waxwork unveiled"
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="https://e3.365dm.com/23/10/70x70/skynews-dwayne-johnson-waxwork_6333809.jpg?20231024132712"
          />
        </div>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-asrneu-MuiTypography-root"
        >
          Also Not Dwayne Johnson 'belly laughs' after controversial waxwork unveiled
        </h6>
        <div
          class="css-urd5lq-MuiStack-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1etgu4m-MuiTypography-root"
            data-testid="description"
          >
            A museum has changed the skin tone of its Dwayne Johnson wax statue after it was criticised for being too white.
          </span>
          <div
            class="css-1jligzp-MuiStack-root"
            data-testid="details"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
              data-testid="time"
            >
              Published 1 week ago
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Publisher1, Publisher1
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Author1, Author2
            </p>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
